river/command/input.zig

```diff
@@ -132,75 +84,26 @@ pub fn input(
-    } else if (mem.eql(u8, "click-method", args[2])) {
-        input_config.click_method = meta.stringToEnum(InputConfig.ClickMethod, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "drag", args[2])) {
-        input_config.drag_state = meta.stringToEnum(InputConfig.DragState, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "drag-lock", args[2])) {
-        input_config.drag_lock = meta.stringToEnum(InputConfig.DragLock, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "disable-while-typing", args[2])) {
-        input_config.dwt_state = meta.stringToEnum(InputConfig.DwtState, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "middle-emulation", args[2])) {
-        input_config.middle_emulation = meta.stringToEnum(InputConfig.MiddleEmulation, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "natural-scroll", args[2])) {
-        input_config.natural_scroll = meta.stringToEnum(InputConfig.NaturalScroll, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "left-handed", args[2])) {
-        input_config.left_handed = meta.stringToEnum(InputConfig.LeftHanded, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "tap", args[2])) {
-        input_config.tap_state = meta.stringToEnum(InputConfig.TapState, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "tap-button-map", args[2])) {
-        input_config.tap_button_map = meta.stringToEnum(InputConfig.TapButtonMap, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "pointer-accel", args[2])) {
-        input_config.pointer_accel = InputConfig.PointerAccel{
-            .value = std.math.clamp(
-                try std.fmt.parseFloat(f32, args[3]),
-                @as(f32, -1.0),
-                @as(f32, 1.0),
-            ),
+        var input_config = InputConfig{
+            .identifier = identifier_owned,
         };
-    } else if (mem.eql(u8, "scroll-method", args[2])) {
-        input_config.scroll_method = meta.stringToEnum(InputConfig.ScrollMethod, args[3]) orelse
-            return Error.UnknownOption;
-    } else if (mem.eql(u8, "scroll-button", args[2])) {
-        const ret = c.libevdev_event_code_from_name(c.EV_KEY, args[3].ptr);
-        if (ret < 1) return Error.InvalidButton;
-        input_config.scroll_button = InputConfig.ScrollButton{ .button = @intCast(ret) };
-    } else {
-        return Error.UnknownCommand;
-    }
+        try input_config.parse(args[2], args[3]);
+
+        try server.input_manager.configs.ensureUnusedCapacity(1);
+        server.input_manager.configs.appendAssumeCapacity(input_config);
+
+        const len = server.input_manager.configs.items.len;
+        break :blk &server.input_manager.configs.items[len - 1];
```

Comment:
This might be marginally cleaner using `addOneAssumeCapacity()` and avoiding the need to index to get the pointer.